### PR TITLE
Add template example back in

### PR DESCRIPTION
**Pull request by surminus**
### Description

This was removed in 337c45a8a4d4d35d64728ef6a47abd0fb9186148, but the commit does not explain why. The PR seems to suggest a change in nested hashes, but does that change accessing behaviour in run_state or make the template incorrect?

When coming to use this resource, I wasn't aware of the `lazy` evaluation (I guess I don't know Chef as well I could), and spent some time figuring out how to put values into a template. I only found this
example had been removed after I'd figured it out.

I don't think it hurts to have the example in the README. If it's now incorrect, can we amend it instead of completely removing it? 

### Issues Resolved

Missing example

### Check List
N/A

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by majormoses** [comment]
Wrong is worse than missing IMHO I retroactively updated the changelog to reflect the breaking changes But essentially we strip the path off the key. Given the following examples it would look something like this:
<details>
<Summary> Expand/Collapse Code </Summary>
```ruby
aws_ssm_parameter_store 'get clear_value' do
  path '/testkitchen/ClearTextString'
  return_key 'clear_value'
  action :get
  aws_access_key node['aws_test']['key_id']
  aws_secret_access_key node['aws_test']['access_key']
end

aws_ssm_parameter_store 'get decrypted_value' do
  path '/testkitchen/EncryptedStringDefaultKey'
  return_key 'decrypted_value'
  with_decryption true
  action :get
  aws_access_key node['aws_test']['key_id']
  aws_secret_access_key node['aws_test']['access_key']
end

aws_ssm_parameter_store 'get decrypted_custom_value' do
  path '/testkitchen/EncryptedStringCustomKey'
  return_key 'decrypted_custom_value'
  with_decryption true
  action :get
  aws_access_key node['aws_test']['key_id']
  aws_secret_access_key node['aws_test']['access_key']
end

aws_ssm_parameter_store 'getParameters' do
  path ['/testkitchen/ClearTextString', '/testkitchen']
  return_key 'parameter_values'
  action :get_parameters
  aws_access_key node['aws_test']['key_id']
  aws_secret_access_key node['aws_test']['access_key']
end

aws_ssm_parameter_store 'getParametersbypath' do
  path '/pathtest/'
  recursive true
  with_decryption true
  return_key 'path_values'
  action :get_parameters_by_path
  aws_access_key node['aws_test']['key_id']
  aws_secret_access_key node['aws_test']['access_key']
end


template '/tmp/file_with_data.txt' do
  source 'file_with_data.txt.erb'
  owner 'ec2-user'
  group 'ec2-user'
  mode '0755'
  sensitive true
  variables lazy {
    {
       clear_value: node.run_state['clear_value'],
       decrypted_custom_value: node.run_state['decrypted_custom_value'],
       decrypted_value: node.run_state['decrypted_value'],
       path1_value: node.run_state['path_values']['path1'],
       path2_value: node.run_state['path_values']['path2'],
       parm1_value: node.run_state['parameter_values']['ClearTextString'],
       parm2_value: node.run_state['parameter_values']['testkitchen'],
    }
  }
end
```
</details>

**Comment by Ishaan28malik** [comment]
@majormoses is it still pending ?

**Comment by majormoses** [comment]
@Ishaan28malik I welcome an updated or new PR

**Review by xorima — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Comment by surminus** [comment]
Sorry I forgot about this until I got notifications about approvals :) if it's wrong I will close and if it's still needed someone can update it correctly.

**Pull request closed without merging**
